contracts/contracts/trade/src/contract.rs

```diff
@@ -676,49 +626,266 @@ fn refund_escrow(
+    assert_trade_state_change_is_valid(
+        trade.state,
+        TradeState::FiatDeposited,
+        TradeState::EscrowDisputed,
+    )
+    .unwrap();
+
+    // Update trade State to TradeState::Disputed and sets arbitrator
+    trade.state = TradeState::EscrowDisputed;
+    let random_seed: u32 = (env.block.time.seconds() % 100) as u32;
+    let arbitrator = get_arbitrator_random(deps.as_ref(), random_seed as usize, trade.fiat.clone());
+    trade.arbitrator = Some(arbitrator.arbitrator);
+    TradeModel::store(deps.storage, &trade).unwrap();
+
+    let res = Response::new()
+        .add_attribute("action", "dispute_escrow")
+        .add_attribute("trade_id", trade.id.clone())
+        .add_attribute("state", trade.state.to_string())
+        .add_attribute("arbitrator", trade.arbitrator.unwrap().to_string());
+
+    Ok(res)
+}
+
+fn settle_dispute(
+    deps: DepsMut,
+    info: MessageInfo,
+    trade_id: String,
+    winner: Addr,
+) -> Result<Response, ContractError> {
+    let mut trade = TradeModel::from_store(deps.storage, &trade_id);
+
+    // Check if caller is the arbitrator of the given trade
+    match &trade.arbitrator {
+        None => {
+            return Err(ContractError::Unauthorized {
+                owner: Addr::unchecked(""),
+                caller: info.sender,
+            })
+        }
+        Some(arbitrator) => {
+            if arbitrator.ne(&info.sender) {
+                return Err(ContractError::Unauthorized {
+                    owner: arbitrator.clone(),
+                    caller: info.sender,
+                });
+            }
+        }
+    }
+
+    // Check if TradeState is EscrowDisputed
+    if TradeState::EscrowDisputed.ne(&trade.state) {
+        return Err(InvalidTradeState {
+            current: trade.state,
+            expected: TradeState::EscrowDisputed,
+        });
+    }
+
+    // Load Offer
+    let offer = load_offer(
+        &deps.querier,
+        trade.offer_id.clone(),
+        trade.offer_contract.to_string(),
+    )
+    .unwrap();
+
+    // Define maker and taker
+    let maker = offer.owner.clone();
+    let taker = if trade.seller.eq(&maker) {
+        trade.buyer.clone()
     } else {
+        trade.seller.clone()
+    };
+
+    // Check if winner is eligible, it must be either maker or taker
+    if winner.eq(&maker) {
+        trade.state = TradeState::SettledForMaker;
+    } else if winner.eq(&taker) {
         trade.state = TradeState::SettledForTaker;
+    } else {
+        return Err(ContractError::InvalidSender {
+            sender: winner,
+            buyer: trade.buyer,
+            seller: trade.seller,
+        });
     }
+    TradeModel::store(deps.storage, &trade).unwrap();
+
     // Pay arbitration fee
-    if arbitration_mode {
-        let fee_rate: Uint128 = Uint128::new(10);
-        let fee_amount = amount.multiply_ratio(Uint128::new(1), fee_rate);
+    let amount = trade.amount.clone();
+    let fee_rate: Uint128 = Uint128::new(10);
```

Comment:
Question:
Shouldn't the arbitration fee rate be a parameter?

Reply:
Addressed in https://github.com/Local-Money/localmoney/pull/80/commits/c3247b03c182cac9116322c32ffd594eff599898